Sources/NeedleFoundation/Internal/DependencyProviderRegistry.swift

```diff
@@ -48,6 +48,19 @@ public class __DependencyProviderRegistry {
         providerFactories[componentPath.hashValue] = dependencyProviderFactory
     }
 
+    /// Unregister the given factory closure with given key.
+    ///
+    /// - note: This method is thread-safe.
+    /// - parameter componentPath: The dependency graph path of the component
+    /// the provider is for.
+    public func unregisterDependencyProviderFactory(`for` componentPath: String) {
```

Comment:
Since this is actually meant to be invoked by manually written code, let's move this out of this "internal" class.

Reply:
Per our discussion, leaving this as is because we need it for use within the external `NeedleFoundationTest` module.